crates/bevy_core/src/name.rs

```diff
@@ -0,0 +1,96 @@
+use bevy_reflect::{Reflect, ReflectComponent};
+use bevy_utils::AHasher;
+use std::{
+    hash::{Hash, Hasher},
+    ops::Deref,
+};
+
+/// Component used to identify a entity. Stores a hash for faster comparisons
+#[derive(Debug, Clone, Reflect)]
+#[reflect(Component)]
+pub struct Name {
+    hash: u64, // TODO: Shouldn't be serialized
+    name: String,
```

Comment:
Could this be `std::borrow::Cow<'static, str>`, or could this whole struct be more general then use:
```rust
type Name = FastCompare<String>;
```

Reply:
Yeah it could

Reply:
Im cool with making this less general for now to cut down on the amount of abstraction. If we find multiple use cases later and it seems worth it, we can always factor this out.